x-pack/plugins/transform/public/app/hooks/use_index_data.ts

```diff
@@ -315,6 +322,22 @@ export const useIndexData = (
 
   const renderCellValue = useRenderCellValue(dataView, pagination, tableItems);
 
+  if (
+    dataGrid.status === INDEX_STATUS.LOADED &&
+    dataViewFields !== undefined &&
+    loadIndexDataStartTime.current !== undefined
+  ) {
+    const loadIndexDataDuration = window.performance.now() - loadIndexDataStartTime.current;
+
+    // Set this to undefined so reporting the metric gets triggered only once.
+    loadIndexDataStartTime.current = undefined;
```

Comment:
nice one! Having a single value for each metric per run seems the best way to go at the moment.